streams/delimiter_stream.ts

```diff
@@ -55,24 +55,47 @@ export class DelimiterStream extends TransformStream<Uint8Array, Uint8Array> {
   #bufs: Uint8Array[] = [];
   #delimiter: Uint8Array;
   #matchIndex = 0;
-  #delimLPS: Uint8Array;
+  #delimLPS: Uint8Array | null;
   #disp: DelimiterDisposition;
 
   constructor(
     delimiter: Uint8Array,
     options?: DelimiterStreamOptions,
   ) {
+    let transform: (
+      chunk: Uint8Array,
+      controller: TransformStreamDefaultController<Uint8Array>,
+    ) => void;
+    const flush = (
+      controller: TransformStreamDefaultController<Uint8Array>,
+    ) => {
+      const bufs = this.#bufs;
+      const length = bufs.length;
+      if (length === 0) {
+        controller.enqueue(new Uint8Array());
+      } else if (length === 1) {
+        controller.enqueue(bufs[0]);
+      } else {
+        controller.enqueue(concat(...bufs));
+      }
+    };
+
+    let lps: Uint8Array | null = null;
+
+    if (delimiter.length === 1) {
+      // Delimiter is a single char
+      transform = (chunk, controller) => this.#handleChar(chunk, controller);
+    } else {
+      transform = (chunk, controller) => this.#handle(chunk, controller);
+      lps = createLPS(delimiter);
+    }
     super({
-      transform: (chunk, controller) => {
-        this.#handle(chunk, controller);
-      },
-      flush: (controller) => {
-        controller.enqueue(concat(...this.#bufs));
-      },
+      transform,
+      flush,
     });
 
     this.#delimiter = delimiter;
-    this.#delimLPS = createLPS(delimiter);
+    this.#delimLPS = lps;
     this.#disp = options?.disposition ?? "discard";
   }
```

Comment:
We can simplify the constructor by doing something like:
```ts
constructor(
  delimiter: Uint8Array,
  options?: DelimiterStreamOptions,
) {
  super({
    transform: (chunk, controller) =>
      delimiter.length === 1
        ? this.#handleChar(chunk, controller)
        : this.#handle(chunk, controller),
    flush: (controller) => this.#flush(controller),
  });

  this.#delimiter = delimiter;
  this.#delimLPS = delimiter.length > 1 ? createLPS(delimiter) : null;
  this.#disp = options?.disposition ?? "discard";
}
```
I think we should also factor out the `flush()` logic into its own separate `#flush()` method to match the style of the rest of the class. WDYT?

Reply:
Gentle ping @aapoalas